File: Strategy/Java/src/FahrenheitToCelsius.java

public class FahrenheitToCelsius implements Converter{

	private final String CELSIUS = "°C (Celsius)";
	private final String FAHRENHEIT = "°F (Fahrenheit)";
	
	@Override
	public float convert(float currentMeasure) {
		float convertedMeasure =  (currentMeasure - 32f)*(5f/9f);
		System.out.println(currentMeasure + FAHRENHEIT + " = " + convertedMeasure + CELSIUS);
		return convertedMeasure;
	}

	
}

File: strategy/en_us/cpp/code/sources/fahrenheit_to_celsius.cpp
#include "fahrenheit_to_celsius.h"

float
FahrenheitToCelsius::convert(float current_measure)
{
	float converted_measure = (current_measure - 32.0) * (5.0/9.0);

	cout << current_measure << FAHRENHEIT << " = " << converted_measure << CELSIUS << "\n";

	return converted_measure;
}
